Reachable unwrap panic in `sys_arch_prctl` when invalid `code` given

### Describe the bug
`do_arch_prctl(...).unwrap()` in `kernel/src/syscall/arch_prctl.rs:21` panics when `do_arch_prctl` returns `Err`.

### To Reproduce
1. Invoke `arch_prctl` with `ARCH_GET_GS (0x1004)` or `ARCH_SET_GS (0x1001)`. These paths return `EINVAL`, which is unwrapped and panics.:
```C
#define _GNU_SOURCE
#include <errno.h>
#include <stdio.h>
#include <sys/syscall.h>
#include <unistd.h>

#ifndef ARCH_SET_GS
#define ARCH_SET_GS 0x1001
#endif
#ifndef ARCH_GET_GS
#define ARCH_GET_GS 0x1004
#endif

static void try_code(long code, unsigned long addr) {
    long ret = syscall(SYS_arch_prctl, code, addr);
    printf("arch_prctl(code=0x%lx, addr=0x%lx) -> ret=%ld errno=%d\n", code, addr, ret, errno);
}

int main(void) {
    try_code(ARCH_GET_GS, 0);
    try_code(ARCH_SET_GS, 0);
    return 0;
}
```
2. Run the compiled program in Asterinas.

### Environment
- Official docker asterinas/asterinas:0.16.2-20251211
- Asterinas version: main 9c4f644bd

### Logs
```
~ # /etc/arch_prctl-gs-panic.c 
[   259.345] ERROR: Uncaught panic:
        called `Result::unwrap()` on an `Err` value: Error { errno: EINVAL, msg: Some("GS cannot be accessed from the user space") }
        at /root/asterinas/kernel/src/syscall/arch_prctl.rs:29
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, is_exited: false, cpu_affinity: AtomicIdSet { bits: [1], phantom: PhantomData<ostd::cpu::id::CpuId> }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31489120 }, fair: FairAttr { weight: 1024, pending_weight: 0, vruntime: 374568954 } } })
Printing stack trace:
   1: fn 0xffffffff88953cf0 - pc 0xffffffff88953d3b / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70ad0;

   2: fn 0xffffffff8804c3d0 - pc 0xffffffff8804cbcd / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70af0;

   3: fn 0xffffffff8804e080 - pc 0xffffffff8804e08a / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70ec0;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70ed0;

   5: fn 0xffffffff88af6f50 - pc 0xffffffff88af6f96 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70ee0;

   6: fn 0xffffffff88af4220 - pc 0xffffffff88af42b6 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70f20;

   7: fn 0xffffffff883b7220 - pc 0xffffffff883b7366 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd70fa0;

   8: fn 0xffffffff882d76f0 - pc 0xffffffff882f8e6a / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd710f0;

   9: fn 0xffffffff882d4190 - pc 0xffffffff882d422a / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd844f0;

  10: fn 0xffffffff8809eaf0 - pc 0xffffffff8809f3a1 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84650;

  11: fn 0xffffffff8809f560 - pc 0xffffffff8809f5be / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84b40;

  12: fn 0xffffffff8809ea60 - pc 0xffffffff8809ea8e / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84cf0;

  13: fn 0xffffffff8809e990 - pc 0xffffffff8809e9d6 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84d20;

  14: fn 0xffffffff8818d0f0 - pc 0xffffffff8818d11c / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84d90;

  15: fn 0xffffffff8823c190 - pc 0xffffffff8823c1cf / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84dd0;

  16: fn 0xffffffff8809f520 - pc 0xffffffff8809f547 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84f40;

  17: fn 0xffffffff8807f720 - pc 0xffffffff8807f73d / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84f80;

  18: fn 0xffffffff889cd540 - pc 0xffffffff889cd565 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84fa0;

  19: fn 0xffffffff889b6d80 - pc 0xffffffff889b6e04 / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd84fd0;

  20: fn 0xffffffff88962a15 - pc 0xffffffff88962a1e / registers:

     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd85000;


     rax 0xffffdfffffd709e8; rdx 0xffffffff889b64f2; rcx  0x10000020100d1cf; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bd6ff0; rbp                0x0; rsp 0xffffdfffffd85008;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) ostd::panic::print_stack_trace
          at /root/asterinas/ostd/src/panic.rs:113
(  2) aster_nix::thread::oops::panic_handler
          at /root/asterinas/kernel/src/thread/oops.rs:130
(  3) __ostd_panic_handler
          at ??:?
(  4) __rustc::rust_begin_unwind
          at 15oleklt2dn278x0pmi5ggj0x:?
(  5) core::panicking::panic_fmt
          at ??:?
(  6) core::result::unwrap_failed
          at ??:?
(  7) aster_nix::syscall::arch_prctl::sys_arch_prctl
          at 8vznn2wp0qo2yew1353o8zxoh:?
(  8) aster_nix::syscall::arch::syscall_dispatch
          at /root/asterinas/kernel/src/syscall/mod.rs:213
(  9) aster_nix::syscall::handle_syscall
          at /root/asterinas/kernel/src/syscall/mod.rs:374
( 10) aster_nix::thread::task::create_new_user_task::{closure#0}
          at /root/asterinas/kernel/src/thread/task.rs:95
( 11) aster_nix::thread::task::create_new_user_task::{closure#1}
          at /root/asterinas/kernel/src/thread/task.rs:126
( 12) unwinding::panicking::catch_unwind::do_call::<aster_nix::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:56
( 13) unwinding::panicking::catch_unwind::<unwinding::panic::RustPanic, (), aster_nix::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:42
( 14) unwinding::panic::catch_unwind::<(), aster_nix::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panic.rs:87
( 15) aster_nix::thread::oops::catch_panics_as_oops::<aster_nix::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/asterinas/kernel/src/thread/oops.rs:55
( 16) aster_nix::thread::task::create_new_user_task::{closure#2}
          at /root/asterinas/kernel/src/thread/task.rs:131
( 17) <aster_nix::thread::task::create_new_user_task::{closure#2} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 18) <alloc::boxed::Box<dyn core::ops::function::FnOnce<(), Output = ()> + core::marker::Send> as core::ops::function::FnOnce<()>>::call_once
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2206
( 19) kernel_task_entry
          at /root/asterinas/ostd/src/task/mod.rs:196
( 20) kernel_task_entry_wrapper
          at ??:?
make: *** [Makefile:308: run_kernel] Error 1
```

--- a/kernel/src/syscall/arch_prctl.rs
+++ b/kernel/src/syscall/arch_prctl.rs
@@ -26,7 +26,7 @@ pub fn sys_arch_prctl(
         "arch_prctl_code: {:?}, addr = 0x{:x}",
         arch_prctl_code, addr
     );
-    let res = do_arch_prctl(arch_prctl_code, addr, user_ctx).unwrap();
+    let res = do_arch_prctl(arch_prctl_code, addr, user_ctx)?;
     Ok(SyscallReturn::Return(res as _))
 }
 

